jumpscale/packages/threebot_deployer/bottle/utils.py

```diff
@@ -211,214 +213,246 @@ def redeploy_threebot_solution(
-            farm_id = deployer.get_pool_farm_id(compute_pool_id)
-            farm = zos._explorer.farms.get(farm_id)
-
-            j.logger.debug(f"saving new threebot local config with uuid {new_solution_uuid}")
-            instance_name = f"threebot_{new_solution_uuid}"
-            user_threebot = USER_THREEBOT_FACTORY.get(instance_name)
-            user_threebot.solution_uuid = new_solution_uuid
-            user_threebot.identity_tid = identity.tid
-            user_threebot.name = solution_name
-            user_threebot.owner_tname = threebot.owner_tname
-            user_threebot.farm_name = farm.name
-            user_threebot.state = ThreebotState.RUNNING
-            user_threebot.continent = farm.location.continent
-            user_threebot.explorer_url = identity.explorer_url
-            user_threebot.subdomain_wid = workload_ids[-4]
-            user_threebot.threebot_container_wid = workload_ids[-3]
-            user_threebot.trc_container_wid = workload_ids[-2]
-            user_threebot.reverse_proxy_wid = workload_ids[-1]
-            user_threebot.save()
-            j.logger.debug(f"threebot local config of uuid {new_solution_uuid} saved")
-            j.logger.debug(f"deleting old threebot local config with uuid {solution_uuid}")
-            USER_THREEBOT_FACTORY.delete(f"threebot_{solution_uuid}")
-            j.logger.debug("deployment successful")
-    return user_threebot
+                        raise DeploymentFailed(
+                            f"Failed to create subdomain {domain} on gateway {gateway.node_id} {workload_ids[-1]}. The resources you paid for will be re-used in your upcoming deployments.",
+                            wid=workload_ids[-1],
+                            identity_name=identity.instance_name,
+                        )
+
+                    test_cert = j.config.get("TEST_CERT")
+                    j.logger.debug("creating backup token")
+                    backup_token = str(j.data.idgenerator.idgenerator.uuid.uuid4())
+                    backup_model.token = backup_token
+                    backup_model.tname = metadata["owner"]
+                    backup_model.save()
+                    j.logger.debug(f"backup token {backup_token} created for tname {backup_model.tname}")
+
+                    environment_vars = {
+                        "SDK_VERSION": new_solution_info["branch"],
+                        "INSTANCE_NAME": new_solution_info["name"],
+                        "THREEBOT_NAME": owner,
+                        "DOMAIN": domain,
+                        "SSHKEY": new_solution_info["public_key"],
+                        "TEST_CERT": "true" if test_cert else "false",
+                        "MARKETPLACE_URL": f"https://{j.sals.nginx.main.websites.threebot_deployer_threebot_deployer_root_proxy_443.domain}/",
+                        "DEFAULT_IDENTITY": "test" if "test" in j.core.identity.me.explorer_url else "main",
```

Comment:
Should the escalation emails env vars be added also as in? https://github.com/threefoldtech/js-sdk/blob/42f41b25e93542cf9f5c0ebfd42476a7402b8dc2/jumpscale/packages/threebot_deployer/chats/threebot.py#L350

Reply:
the entrypoint has been updated to save email config before backup. so no need to pass this info again